### PR TITLE
fix: add default mapping and integ tests for addons

### PR DESCRIPTION
**Pull request by bvtujo**
This PR adds a default mapping for the DBMaxCapacity and DBMinCapacity values which doesn't depend on customers having run `env init` before `storage init`. The default value `All` can't collide with an environment name because it contains a capital letter and this is disallowed at env creation. 

I've also added integ tests to check the outputs of the addons `BinaryMarshaler`s. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/addon/storage.go [hunk 1]
```diff
@@ -155,6 +155,7 @@ func (r *RDS) MarshalBinary() ([]byte, error) {
 
 // NewRDS creates a new RDS marshaler which can be used to write CF via addonWriter.
 func NewRDS(input RDSProps) *RDS {
+	input.Envs = append(input.Envs, "All") // Append the special "All" key for Autoscaling mappings
```
Comment (Lou1415926) [inline internal/pkg/addon/storage.go]: I wonder if we should hardcode this into the template instead of adding a dummy env here?

**Review by Lou1415926 — COMMENTED** [review]
Looks good! One concern about adding the "All" env ⬇️

File: internal/pkg/addon/storage.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/addon/storage.go]: Agreed!

File: templates/addons/aurora/cf.yml [hunk 2]
```diff
@@ -30,6 +30,9 @@ Mappings:
       "DBMaxCapacity": 8 # AllowedValues: [2, 4, 8, 16, 32, 64, 192, 384]
       {{end -}}
     {{end}}
+    # Replace DBMinCapacity or DBMaxCapacity below with
```
Comment (efekarakus) [inline templates/addons/aurora/cf.yml]: Should we move this comment closer to the field `MinCapacity`?

File: internal/pkg/addon/addon_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// +build integration localintegration
```
Comment (efekarakus) [inline internal/pkg/addon/addon_integration_test.go:1]: Yay thank you for adding an integ test !! 🌈

**Review by efekarakus — COMMENTED** [review]
Looks great, +1 to Wanxian's comment

**Review by efekarakus — APPROVED** [review]

**Review by Lou1415926 — APPROVED** [review]
Thank you very much for fixing this!!!!

**Pull request merged**